src/metric/regression_metric.hpp

```diff
@@ -162,5 +162,23 @@ class FairLossMetric: public RegressionMetric<FairLossMetric> {
   }
 };
 
+/*! \brief Poisson regression loss for regression task */
+class PoissonMetric: public RegressionMetric<PoissonMetric> {
+public:
+  explicit PoissonMetric(const MetricConfig& config) :RegressionMetric<PoissonMetric>(config) {
+  }
+
+  inline static double LossOnPoint(float label, double score, double, double) {
+    const double eps = 1e-10f;
+    if (score < eps) {
+      score = eps;
```

Comment:
yes. this is because score should be positive.
cf. score is:
![image](https://cloud.githubusercontent.com/assets/433434/22401553/993483ec-e61b-11e6-9094-e2708fad927d.png)
(from https://en.wikipedia.org/wiki/Poisson_regression)